### PR TITLE
TensorMechanics action block restriction fixes

### PR DESCRIPTION
**Pull request by dschwen**
This fixes:

* variable generation for multiple TM actions with different block restrictions
* tensor mechanics actions operating on blocks with different coordinate systems

Refs #7987

### CODE REVIEW

**Review by bwspenc — APPROVED** [review]
Looks good.  This is a big improvement.

File: modules/tensor_mechanics/src/actions/TensorMechanicsAction.C [hunk 1]
```diff
@@ -332,6 +353,31 @@ TensorMechanicsAction::actOutputGeneration()
...
+  // Gather info about all other master actions when we add variables
+  //
+  if (_current_task == "validate_coordinate_systems" && getParam<bool>("add_variables"))
+  {
+    auto actions = _awh.getActions<TensorMechanicsAction>();
+    for (const auto & action : actions)
+    {
+      const auto size_before = _subdomain_id_union.size();
+      const auto added_size = action->_subdomain_ids.size();
+      _subdomain_id_union.insert(action->_subdomain_ids.begin(), action->_subdomain_ids.end());
```
Comment (permcody) [inline modules/tensor_mechanics/src/actions/TensorMechanicsAction.C]: Interesting, any reason you don't use one of the STL "set" operations directly? That's what you are doing semantically.

File: modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C [hunk 2]
```diff
@@ -24,28 +24,23 @@ InputParameters validParams<StressDivergenceRSphericalTensors>()
...
 StressDivergenceRSphericalTensors::computeQpResidual()
 {
-  mooseAssert(_component == 0, "Invalid component for this 1D RSpherical problem.");
-
-  Real div = _grad_test[_i][_qp](0) * _stress[_qp](0,0) + //stress_{rr} part 1
-    + ( _test[_i][_qp] / _q_point[_qp](0)) * _stress[_qp](1,1) + //stress_{\theta \theta}
-    + ( _test[_i][_qp] / _q_point[_qp](0)) * _stress[_qp](2,2); //stress_{\phi \phi}
-
-  return div;
+  return   _grad_test[_i][_qp](0) * _stress[_qp](0,0) +               //stress_{rr} part 1
```
Comment (permcody) [inline modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C]: 3 spaces after `return` keyword

File: modules/tensor_mechanics/src/actions/TensorMechanicsAction.C (hunk 1 above)
Comment (dschwen) [inline modules/tensor_mechanics/src/actions/TensorMechanicsAction.C]: I'm doing three things here:

1. creating a set union
2. making sure the original sets are not overlapping
3. making sure none of the sets in the union is empty

I'm open to suggestions on how to make this more compact with STL.

File: modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C (hunk 2 above)
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C]: Yep, to line up the following terms. I did introduce duplicate `+` operators though.

**Pull request merged**